src/plugins/controls/public/options_list/components/options_list_popover_suggestions.tsx

```diff
@@ -49,130 +43,95 @@ export const OptionsListPopoverSuggestions = ({
-              return;
-            }
-            if (selectedOptionsSet.has(key)) {
-              dispatch(deselectOption(key));
-              return;
-            }
-            dispatch(selectOption(key));
-          }}
-          className={
-            showOnlySelected && invalidSelectionsSet.has(key)
-              ? 'optionsList__selectionInvalid'
-              : 'optionsList__validSuggestion'
-          }
-          aria-label={
-            availableOptions?.[key]
-              ? OptionsListStrings.popover.getSuggestionAriaLabel(
-                  key,
-                  availableOptions[key].doc_count ?? 0
-                )
-              : key
-          }
-        >
-          <EuiFlexGroup justifyContent="spaceBetween" responsive={false} alignItems="center">
-            <EuiFlexItem
-              grow={false}
-              css={css`
-                min-width: 0;
-              `}
-            >
-              <span
-                css={css`
-                  overflow: hidden;
-                  text-overflow: ellipsis;
-                `}
-              >{`${key}`}</span>
-            </EuiFlexItem>
-            {!showOnlySelected && (
-              <EuiFlexItem grow={false} tabIndex={-1}>
-                {availableOptions && availableOptions[key] && (
-                  <EuiToolTip
-                    content={OptionsListStrings.popover.getDocumentCountTooltip(
-                      availableOptions[key].doc_count
-                    )}
-                    position={'right'}
-                  >
-                    <EuiText
-                      className="eui-textNumber"
-                      size="xs"
-                      color={euiTheme.colors.subduedText}
-                      css={css`
-                        font-weight: ${euiTheme.font.weight.medium};
-                      `}
-                    >
-                      {`${availableOptions[key].doc_count.toLocaleString()}`}
-                    </EuiText>
-                  </EuiToolTip>
-                )}
-              </EuiFlexItem>
-            )}
-          </EuiFlexGroup>
-        </EuiFilterSelectItem>
-      ))}
-    </>
+    <EuiSelectable
+      isLoading={isLoading}
+      options={suggestionsTest}
+      listProps={{ onFocusBadge: false }}
+      aria-label={OptionsListStrings.popover.getSuggestionsAriaLabel(fieldName)}
+      emptyMessage={<OptionsListPopoverEmptyMessage showOnlySelected={showOnlySelected} />}
+      onChange={(newSuggestions, _, changedOption) => {
+        setSuggestionsTest(newSuggestions);
+
+        const key = changedOption.key ?? changedOption.label;
+        if (key === 'exists-option') {
+          dispatch(selectExists(!Boolean(existsSelected)));
+        } else if (showOnlySelected) {
+          dispatch(deselectOption(key));
+        } else if (selectedOptionsSet.has(key)) {
+          dispatch(deselectOption(key));
+        } else if (singleSelect) {
+          dispatch(replaceSelection(key));
```

Comment:
Note that the `singleSelect` check was moved down one level because the previous order of checks was causing a bug where, when `singleSelect` was on, selections could not be undone unless `showOnlySelected` was `true`.